runtime/src/bank.rs

```diff
@@ -4127,6 +4135,35 @@ impl Bank {
         let _ = self.executor_cache.write().unwrap().remove(pubkey);
     }
 
+    fn program_modification_slot(&self, pubkey: &Pubkey) -> Result<Slot> {
+        let program = self
+            .get_account_with_fixed_root(pubkey)
+            .ok_or(TransactionError::ProgramAccountNotFound)?;
+        if bpf_loader_upgradeable::check_id(program.owner()) {
+            if let Ok(UpgradeableLoaderState::Program {
+                programdata_address,
+            }) = program.state()
+            {
+                let programdata = self
+                    .get_account_with_fixed_root(&programdata_address)
+                    .ok_or(TransactionError::ProgramAccountNotFound)?;
+                if let Ok(UpgradeableLoaderState::ProgramData {
+                    slot,
+                    upgrade_authority_address: _,
+                }) = programdata.state()
+                {
+                    return Ok(slot);
+                }
+            }
+            return Err(TransactionError::ProgramAccountNotFound);
+        } else if loader_v3::check_id(program.owner()) {
+            let state = solana_loader_v3_program::get_state(program.data())
+                .map_err(|_| TransactionError::ProgramAccountNotFound)?;
+            return Ok(state.slot);
+        }
```

Comment:
There should be an `else` case for the deprecated bpf loader and the normal bpf loader.

Reply:
It falls through to return `Ok(0)` for the else case at the end of the function.

Reply:
Yes, but maybe an explicit `else` with a comment that this handles the other two bpf loaders?